portafly/src/components/data-list/modals/ChangeStateModal.tsx

```diff
@@ -39,40 +37,37 @@ const ChangeStateModal: React.FunctionComponent<Props> = ({
         addAlert({ id: 'success', variant: 'success', title: t('toasts.change_state_start') })
       })
       .catch(() => {
-        actionFailed()
-        addAlert({ id: 'error', variant: 'danger', title: t('toasts.change_state_error') })
+        const error = t('toasts.change_state_error')
+        actionFailed(error)
       })
   }
 
-  const actions = [
-    <Button
-      key="confirm"
-      variant="primary"
+  const submitButton = (
+    <SubmitButton
+      key="submit"
       onClick={onSubmit}
       isDisabled={value === '' || isLoading}
     >
       {t('modals.change_state.send')}
-    </Button>,
-    <Button key="cancel" variant="link" onClick={onClose}>
-      {t('modals.change_state.cancel')}
-    </Button>
-  ]
+    </SubmitButton>
+  )
 
   const options = [
     { value: '', label: '' },
-    { value: 'approved', label: t('state.approved') },
-    { value: 'pending', label: t('state.pending') },
-    { value: 'rejected', label: t('state.rejected') },
-    { value: 'suspended', label: t('state.suspended') }
+    { value: 'approved', label: t('states.approved') },
+    { value: 'pending', label: t('states.pending') },
+    { value: 'rejected', label: t('states.rejected') },
+    { value: 'suspended', label: t('states.suspended') }
   ]
 
   return (
     <DataListModal
       title={t('modals.change_state.title')}
-      onClose={onClose}
-      actions={actions}
+      submitButton={submitButton}
       items={items}
       to={t('modals.change_state.to')}
+      errorMsg={errorMsg}
+      shouldWarnClose={value !== ''}
```

Comment:
What is the meaning of comparing to empty string here and in https://github.com/3scale/porta/pull/1920/files#diff-f4365cf4bc3becc85a5085ad7f7793d2R49 ?

Reply:
Here it means it warns the user when closing if there's anything written.
There it means it is disabled as long as the form is empty (form validation).